.github/workflows/legend-release.yml

```diff
@@ -0,0 +1,373 @@
+              echo "Error: Legend Pure to be released without providing versions for all its dependent projects."
+              exit 1
+            fi
+          fi
+          if [ -n "${{ github.event.inputs.legend-engine-release-version }}" ]; then
+            if [ -z "${{ github.event.inputs.legend-sdlc-release-version }}" ] && [ -z "${{ github.event.inputs.legend-depot-release-version }}" ]; then
+              echo "Error: Legend Engine to be released without providing versions for all its dependent projects."
+              exit 1
+            fi
+          fi
+          if [ -n "${{ github.event.inputs.legend-sdlc-release-version }}" ]; then
+            if [ -z "${{ github.event.inputs.legend-depot-release-version }}" ]; then
+              echo "Error: Legend SDLC to be released without providing versions for all its dependent projects."
+              exit 1
+            fi
+          fi
+
+
+      # Legend Shared Release
+      - name: Set Required RunNumber of Legend Shared Release
+        if: success() && github.event.inputs.legend-shared-release-version
+        run: |
+          auth_token="${{ env.RELEASE_TOKEN }}"
+          owner="${{ env.OWNER }}"
+          repo="legend-shared"
+          workflow_id="${{ env.WORKFLOW_ID }}"
+  
+          # Get the last run number
+          lastRunNumber=$(curl -s -H "Authorization: token $auth_token" "https://api.github.com/repos/$owner/$repo/actions/workflows/$workflow_id/runs" | jq -r '.workflow_runs[0].run_number')
+  
+          # Calculate the required run number
+          requiredRunNumber=$((lastRunNumber + 1))
+  
+          # Set the environment variable
+          echo "Setting LEGEND_SHARED_RUN_NUMBER to $requiredRunNumber"
+          echo "LEGEND_SHARED_RUN_NUMBER=${requiredRunNumber}" >> $GITHUB_ENV
+      - name: Trigger Legend Shared Release
+        if: success() && github.event.inputs.legend-shared-release-version
+        uses: peter-evans/repository-dispatch@v1
+        with:
+          token: ${{ env.RELEASE_TOKEN }}
+          repository: ${{ env.OWNER }}/legend-shared
+          event-type: ${{ env.EVENT_TYPE }}
+          client-payload: '{"releaseVersion": "${{ github.event.inputs.legend-shared-release-version }}"}'
+      - name: Check Status Of Legend Shared Release
+        if: success() && github.event.inputs.legend-shared-release-version
+        run: |
+          sleep 60                                # Wait for 1 minute before polling the API again
+
+          auth_token="${{ env.RELEASE_TOKEN }}"
+          owner="${{ env.OWNER }}"
+          repo="legend-shared"
+          workflow_id="${{ env.WORKFLOW_ID }}"
+          required_run_number="${{ env.LEGEND_SHARED_RUN_NUMBER }}"
+          
+          # Find the ID of the required workflow run
+          run_id=$(curl -s -H "Authorization: token $auth_token" \
```

Comment:
Can't we just fetch the final entry in the 'workflow_runs' array and avoid having to calculate $required_run_number?
The run we're interested in will always be the last one right? That's what we already calculate via 'lastRunNumber'?
https://github.com/stedolan/jq/issues/509

Reply:
I did this to avoid race condition and scenario's where something went wrong